src/webpack/CompilerEventsFacade.ts

```diff
@@ -5,36 +5,35 @@ export default class CompilerEventsFacade {
 
   private _compiler: Compiler;
 
-  private _legacyTapable: boolean;
-
   constructor(compiler: Compiler) {
     this._compiler = compiler;
-    this._legacyTapable = !compiler.hooks;
   }
 
-  public afterOptimizeChunkAssets(
+  public afterOptimizeChunks(
     call: (compilation: Compilation, chunks: Compilation["chunks"]) => void,
   ) {
-    return this._legacyTapable
-      ? (this._compiler as any).plugin("compilation",comp =>
-          comp.plugin("after-optimize-chunk-assets",chunks =>
-            call(comp, chunks),
-          ),
-        )
-      : this._compiler.hooks.compilation.tap(
-          CompilerEventsFacade.extensionName,comp =>
-            comp.hooks.afterOptimizeChunkAssets.tap(
-              CompilerEventsFacade.extensionName,chunks => call(comp, chunks),
-            ),
+    return this._compiler.hooks.compilation.tap(
+      CompilerEventsFacade.extensionName,
+      (comp) => {
+        const afterOptimizeChunkAssets = (chunks) => {
+          call(comp, chunks);
+        };
+        /* https://github.com/webpack/webpack/blob/main/lib/Compilation.js#L772-L779
+        afterOptimizeChunkAssets = PROCESS_ASSETS_STAGE_OPTIMIZE + 1
+        */
+        const stage = Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE + 1;
+        (comp.hooks as any).processAssets.tap(
+          { name: CompilerEventsFacade.extensionName, stage },
+          afterOptimizeChunkAssets,
```

Comment:
Let's release this as a minor patch, and work on refactoring this to use the new `processAssets` and release that as a major patch.